libraries/AP_AerobridgeTrustedFlight/AP_AerobridgeTrustedFlight.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: @rhythmize
+ */
+#include <AP_Filesystem/AP_Filesystem.h>
+#include <GCS_MAVLink/GCS.h>
+#include <mbedtls/pem.h>
+#include "AP_AerobridgeTrustedFlight.h"
+
+extern const AP_HAL::HAL& hal;
+
+AP_AerobridgeTrustedFlight::AP_AerobridgeTrustedFlight()
+{
+    if (_singleton != nullptr) {
+        AP_HAL::panic("Too many AerobridgeTrustedFlight modules");
+        return;
+    }
+
+    _singleton = this;
+}
+
+bool AP_AerobridgeTrustedFlight::is_trusted()
```

Comment:
this should look like this: ```
bool AP_AerobridgeTrustedFlight::is_trusted(size_t buflen, char *buffer)
{
  if (!good) {
  hal.util->snprintf(buffer, buflen, "No good - try again tomorrow");
}
}

Reply:
@tridge What would be the best place to write debug/ info logs, if any? Or all the logs should go to the same buffer for displaying error messages? Also, since this is a one time pre-arm check should we log the success message for visibility or ignore it?

Reply:
resolved